src/common/procedure/src/local/runner.rs

```diff
@@ -470,8 +476,18 @@ impl Runner {
     }
 
     async fn rollback_procedure(&mut self) -> Result<()> {
```

Comment:
It only writes the state to the store.
```suggestion
    async fn write_procedure_state(&mut self) -> Result<()> {
```